openslides_backend/action/actions/user/user_scope_permission_check_mixin.py

```diff
@@ -0,0 +1,94 @@
+from enum import Enum, auto
+from typing import Any, Dict, Tuple
+
+from ....permissions.management_levels import (
+    CommitteeManagementLevel,
+    OrganisationManagementLevel,
+)
+from ....permissions.permission_helper import (
+    has_committee_management_level,
+    has_organisation_management_level,
+    has_perm,
+)
+from ....permissions.permissions import Permissions
+from ....services.datastore.interface import GetManyRequest
+from ....shared.exceptions import MissingPermission
+from ....shared.patterns import Collection, FullQualifiedId
+from ...action import Action
+
+
+class UserScope(Enum):
+    Meeting = auto()
+    Committee = auto()
+    Organisation = auto()
+
+
+class UserScopePermissionCheckMixin(Action):
+    def check_permissions_for_scope(self, instance: Dict[str, Any]) -> None:
+        """
+        Checks the permissions for user-altering actions depending on the user scope.
+        """
+        if has_organisation_management_level(
+            self.datastore, self.user_id, OrganisationManagementLevel.CAN_MANAGE_USERS
+        ):
+            return
+
+        scope, scope_id = self.get_user_scope(instance["id"])
+        if scope == UserScope.Committee:
+            if not has_committee_management_level(
+                self.datastore,
+                self.user_id,
+                CommitteeManagementLevel.CAN_MANAGE,
+                scope_id,
+            ):
+                raise MissingPermission(
+                    {
+                        OrganisationManagementLevel.CAN_MANAGE_USERS: 1,
+                        CommitteeManagementLevel.CAN_MANAGE: scope_id,
+                    }
+                )
+        elif scope == UserScope.Meeting:
+            meeting = self.datastore.fetch_model(
+                FullQualifiedId(Collection("meeting"), scope_id), ["committee_id"]
+            )
+            if not has_committee_management_level(
+                self.datastore,
+                self.user_id,
+                CommitteeManagementLevel.CAN_MANAGE,
+                meeting["committee_id"],
+            ) and not has_perm(
+                self.datastore, self.user_id, Permissions.User.CAN_MANAGE, scope_id
+            ):
+                raise MissingPermission(
+                    {
+                        OrganisationManagementLevel.CAN_MANAGE_USERS: 1,
+                        CommitteeManagementLevel.CAN_MANAGE: meeting["committee_id"],
+                        Permissions.User.CAN_CHANGE_OWN_PASSWORD: scope_id,
```

Comment:
May be this should be Permissions.User.CAN_MANAGE

Reply:
Yes, you are of course correct, will fix it